test/simplifyExpression/basicsSearch/basicTest.js

```diff
@@ -0,0 +1,148 @@
+const assert = require('assert');
+
+const {parse, print} = require('math-parser');
+
+const basic_rules = require('../../../lib/simplifyExpression/basicsSearch/basicRules');
+
+describe('basic rules', function() {
+  describe('negation', function() {
+    const tests = [
+      ['--1','1'],
+      ['--x','x'],
+      ['--(x + 1)', 'x + 1'],
+    ];
+    tests.forEach(t => testSimplify(t[0], t[1], basic_rules.negation));
+  });
+
+  describe.skip('rearrange coefficient', function() {
+    const tests = [
+        ['y^3 * 5', '5 y^3'],
+        ['yz * 3', '3 yz'],
+        // TODO: handle this case better
+        //['3x^2 * 5', '5 (3 x^2)']
+    ];
+    tests.forEach(t => testSimplify(t[0], t[1], basic_rules.REARRANGE_COEFF));
+  });
+
+  describe('division by negative one', function() {
+    const tests = [
+      ['2 / -1','-2'],
+      ['x / -1','-x'],
+      ['(x + 1) / -1', '-(x + 1)'],
+    ];
+    tests.forEach(t => testSimplify(t[0], t[1], basic_rules.removeDivisionByOne));
+  });
+
+  describe('division by one', function() {
+    const tests = [
+      ['2 / 1', '2'],
+      ['x / 1', 'x'],
+      ['(x + 1) / 1', 'x + 1'],
+    ];
+    tests.forEach(t => testSimplify(t[0], t[1], basic_rules.removeDivisionByOne));
+  });
+
+  describe('multiply by zero', function() {
+    const tests = [
+      ['2 * 0', '0'],
+      ['x * 0', '0'],
+      ['x 0', '0'],
+      ['(x + 1) * 0', '0'],
+      // reverse
+      ['0 * 2', '0'],
+      ['0 * x', '0'],
+      ['0 x', '0'],
+      ['0 * (x + 1)', '0'],
+    ];
+    tests.forEach(t => testSimplify(t[0], t[1], basic_rules.removeMultiplicationByZero));
+  });
+
+  describe('reduce exponent by zero', function() {
+    const tests = [
+      ['2 ^ 0', '1'],
```

Comment:
It would be parsed correctly either way.  Also, `math-parser`'s `print` function doesn't add spaces around the exponent operator, see https://github.com/semantic-math/math-parser/blob/master/lib/__test__/print.test.js#L53-L64.